torch_xla/csrc/ops/device_data.cpp

```diff
@@ -13,16 +13,34 @@ DeviceData::DeviceData(std::shared_ptr<torch::lazy::BackendData> data)
               /*hash_seed=*/(uint32_t)101),
       data_(std::move(data)) {}
 
+DeviceData::DeviceData(std::shared_ptr<torch::lazy::BackendData> data,
+                       torch::lazy::OpList ops, xla::Shape xla_shape,
+                       xla::OpSharding sharding)
+    : XlaNode(xla_device_data, ops, {data->shape()}, xla_shape, sharding,
```

Comment:
Do we need `xla_shape` as input? Could we do this like in the other constructor `UnwrapXlaData(data)->shape()`?

Reply:
I can experiment with it. Any reason to exclude xla_shape?

Reply:
Generally, if it's not needed or redundant, then I would not add more arguments to the constructor.